src/libappimage/utils/ElfFile.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+// system
+#include <string>
+#include <cstring>
+
+// local
+#include "light_elf.h"
+
+namespace appimage {
+    namespace utils {
+        /**
+         * Utility class to read elf files. Not meant to be feature complete
+         */
+        class ElfFile {
+        public:
+            explicit ElfFile(const std::string& path);
+
+
+            /*
+             * Calculate the size of an ELF file on disk based on the information in its header
+             *
+             * Example:
+             *
+             * ls -l   126584
+             *
+             * Calculation using the values also reported by readelf -h:
+             * Start of section headers	e_shoff		124728
+             * Size of section headers		e_shentsize	64
+             * Number of section headers	e_shnum		29
+             *
+             * e_shoff + ( e_shentsize * e_shnum ) =	126584
+             */
+            ssize_t getSize();
+
+        private:
```

Comment:
PImpl. Also makes them easier to hide from the global interface in the future, which is something I _really_ want to accomplish. Might be worth opening an issue...